compiler_opt/tools/regalloc_trace/compile_corpus.py

```diff
@@ -0,0 +1,92 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Compiles a corpus for further downstream processing."""
+
+import time
+
+from absl import app
+from absl import flags
+from absl import logging
+import gin
+
+from compiler_opt.rl import corpus
+from compiler_opt.es.regalloc_trace import regalloc_trace_worker
+from compiler_opt.rl import registry
+
+_CORPUS_PATH = flags.DEFINE_string(
+    'corpus_path', None, 'The path to the corpus.', required=True)
+_OUTPUT_PATH = flags.DEFINE_string(
+    'output_path', None, 'The path to the output.', required=True)
+_GIN_FILES = flags.DEFINE_multi_string(
+    'gin_files', [], 'List of paths to gin configuration files to use.')
+_GIN_BINDINGS = flags.DEFINE_multi_string(
+    'gin_bindings', [], 'Gin bindings to override the values in config files.')
+_TFLITE_POLICY_PATH = flags.DEFINE_string(
+    'tflite_policy_path', None,
+    'The path to the TFLite policy to use, if there is one.')
+_MODE = flags.DEFINE_enum('mode', 'full', ['full', 'bc', 'asm'],
```

Comment:
do you need all these for your scenario right now?

Reply:
Technically we do not need the `asm` mode, but it makes testing that the extracted corpus and serialized BBs are actually able to supply all the data needed for modelling the traces much easier. `full` and `bc` are definitely needed to serialize BBs and to generate the post middle end bitcode.